dali/operators/generic/one_hot.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2017-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/generic/one_hot.h"
+#include "dali/core/tensor_shape.h"
+
+namespace dali {
+
+DALI_REGISTER_OPERATOR(OneHot, OneHot, CPU);
+
+DALI_SCHEMA(OneHot)
+    .DocStr(
+        "Produce tensor representing one hot encoding "
+        " of the given input")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddOptionalArg("depth", R"code(Number of all classes)code", 0)
+    .AddOptionalArg(arg_names::kDtype, R"code(Data type for the output)code",
+                    DALI_FLOAT)
+    .AddOptionalArg("on_value", R"code(Value that will be used to fill the output when input[j] = i)code", 1.f)
+    .AddOptionalArg("off_value", R"code(Value that will be used to fill the output when input[j] != i)code", 0.f);
+
+bool OneHot::SetupImpl(std::vector<OutputDesc> &output_desc, const HostWorkspace &ws) {
+  output_desc.resize(1);
+  const auto &input = ws.template InputRef<CPUBackend>(0);
+  output_desc[0].shape = uniform_list_shape(batch_size_, (shape_cat(input[0].shape(), depth_)));
```

Comment:
This assumes that `input[sample].shape` is equal for all samples - this is generally not guaranteed and not required.

Reply:
I have put a DALI_ENFORCE clause here that will validate the input